EmbeddedPkg/Application/AndroidFastboot/AndroidFastbootApp.c

```diff
@@ -160,15 +162,15 @@ HandleDownload (
     SEND_LITERAL ("FAILNot enough memory");
   } else {
     ZeroMem (Response, sizeof Response);
-    if (mTransport->RequestReceive) {
-      mTransport->RequestReceive (mNumDataBytes);
-    }
     AsciiSPrint (Response, sizeof Response, "DATA%x",
       (UINT32)mNumDataBytes);
     mTransport->Send (sizeof Response - 1, Response, &mFatalSendErrorEvent);
 
     mState = ExpectDataState;
     mBytesReceivedSoFar = 0;
+    if (mTransport->RequestReceive) {
+      mTransport->RequestReceive (mNumDataBytes);
+    }

```

Comment:
@chenyu56 Your comments didn't hit the point. You change the sequence at here. You give a response first before receiving data. Could you explain the reason?

Reply:
Moving these codes after sending response to host and set  "mBytesReceivedSoFar = 0" is more reasonable.